rust/chains/hyperlane-cosmos/src/providers/grpc.rs

```diff
@@ -534,15 +521,43 @@ impl WasmProvider for WasmGrpcProvider {
         Ok(response.data)
     }
 
+    async fn wasm_contract_info(&self) -> ChainResult<ContractInfo> {
+        let contract_address = self.get_contract_address()?;
+        let response = self
+            .provider
+            .call(move |provider| {
+                let to = contract_address.address().clone();
+                let future = async move {
+                    let mut client = WasmQueryClient::new(provider.channel.clone());
+
+                    let request = tonic::Request::new(QueryContractInfoRequest { address: to });
+
+                    let response = client
+                        .contract_info(request)
+                        .await
+                        .map_err(ChainCommunicationError::from_other)?
+                        .into_inner()
+                        .contract_info
+                        .ok_or(ChainCommunicationError::from_other_str(
```

Comment:
wonder if this function returning a `ChainResult<Option<ContractInfo>>` would be possible / more clear as a return type so we can disambiguate genuine gRPC/network errors from the provided address not being a contract

That is, assuming their API doesn't give an error if there's no contract and instead it has contract_info here as None

Reply:
I tested it. Their API returns an error if an address is not a contract. 

So, we return an error in two cases: address is definitely not a contract (first error) and if `contract_info` is empty (it may mean that contract does not have the code to process the asset we are going to transfer to it).

It is strange that a contract account may have empty `contract_info`. I did not see such a case in my testing.

I suppose that `contract_info` is `Option<ContractInfo>` because optional fields is the best practice in gRPC/protobuf definitions, but it is just a conjecture.